### PR TITLE
Pure phantom

### PR DESCRIPTION
**Pull request by EWhite613**
#major#

#new-baseline#

### CODE REVIEW

**Comment by EWhite613** [comment]
Great PhantomJS never exits on Linux...

**Comment by EWhite613** [comment]
https://github.com/testem/testem/issues/915

**Comment by ember-cli-visual-acceptance** [comment]
PR ember-cli-visual-acceptance Report 
 [Imgur Album](http://imgur.com/a/1WFF2)

**Comment by EWhite613** [comment]
@sglanzer With some weird magic described here https://github.com/testem/testem/issues/915#issuecomment-231097612 I somehow got Testem to work with our custom Launcher.

Ready for a review

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ branches:
   except:
   - /^v[0-9\.]+/
 before_install:
-- npm install -g coveralls pr-bumper
+- npm install -g coveralls pr-bumper phantomjs slimerjs
```
Comment (sglanzer) [inline .travis.yml]: Are we using phantom or slimer?  Seems like one should go

File: testem.js [hunk 2]
```diff
@@ -1,16 +1,22 @@
 module.exports = {
   'framework': 'mocha',
   'test_page': 'tests/index.html?hidepassed',
+  'on_exit': 'echo Dude we are closed',
```
Comment (sglanzer) [inline testem.js]: Test text? ;)

File: tests/integration/simple-test.js [hunk 3]
```diff
@@ -34,5 +34,23 @@ describeComponent(
...
+        svg { background:#fff; }
+        .face { stroke:#000; stroke-width:20px; stroke-linecap:round }
+      </style>
+      <circle r="200" class="face" fill="red"/>
+      <path fill="none" class="face" transform="translate(-396,-230)" d="M487,282c-15,36-51,62-92,62 c-41,0-77-25-92-61"/>
+      <circle cx="-60" cy="-50" r="20" fill="#000"/>
+      <circle cx="60" cy="-50" r="20" fill="#000"/>
+    </svg>`)
+      capture('svg').then(function (data) {
+        console.log(arguments)
```
Comment (sglanzer) [inline tests/integration/simple-test.js]: Missed log?

File: vendor/VisualAcceptance.js [hunk 4]
```diff
@@ -76,24 +74,23 @@ function capture(imageName, width, height, misMatchPercentageMargin, assert) {
       })
       $(document.getElementById('ember-testing')).removeAttr('style')
       $(document.getElementById('ember-testing-container')).removeAttr('style')
-      node.innerHTML = '<div class="test pass"> <div class="list-name"> No passed image. Saving current as baseline: ' + imageName + '</div> <div class="additional-info"> Addition Information: </div> <img src="'+ image + '" /> </div>'
+      node.innerHTML = '<div class="test pass"> <div class="list-name"> No passed image. Saving current as baseline: ' + imageName + '</div> <div class="additional-info"> Addition Information: </div> <img src="' + image + '" /> </div>'
```
Comment (sglanzer) [inline vendor/VisualAcceptance.js]: New

File: vendor/VisualAcceptance.js [hunk 5]
```diff
@@ -106,7 +103,7 @@ function capture(imageName, width, height, misMatchPercentageMargin, assert) {
               }
             })
             result = true
-            node.innerHTML = '<div class="test pass"> <div class="list-name">  Passed: ' + imageName + '</div> <div class="additional-info"> Addition Information: </div> <img src="'+ image + '" /> </div>'
+            node.innerHTML = '<div class="test pass"> <div class="list-name">  Passed: ' + imageName + '</div> <div class="additional-info"> Addition Information: </div> <img src="' + image + '" /> </div>'
```
Comment (sglanzer) [inline vendor/VisualAcceptance.js]: We don't need to include passed in the report

File: vendor/VisualAcceptance.js [hunk 6]
```diff
@@ -118,24 +115,25 @@ function capture(imageName, width, height, misMatchPercentageMargin, assert) {
                 name: browserDirectory + imageName + '.png'
               }
             })
-            node.innerHTML = '<div class="test fail"> <div class="list-name">  Failed: '+ imageName+' </div> <div class="additional-info"> Addition Information: </div> <div class="images"> <div class="image"> <img class="diff" src="'+data.getImageDataUrl()+'" /> <div class="caption">  Diff   </div> </div> <div class="image">  <img class="input" src="'+image+'" /> <div class="caption"> Current  </div> </div> <div class="image"> <img class="passed" src="'+res.image+'" /> <div class="caption"> Baseline   </div> </div> </div> </div>'
+            node.innerHTML = '<div class="test fail"> <div class="list-name">  Failed: ' + imageName + ' </div> <div class="additional-info"> Addition Information: </div> <div class="images"> <div class="image"> <img class="diff" src="' + data.getImageDataUrl() + '" /> <div class="caption">  Diff   </div> </div> <div class="image">  <img class="input" src="' + image + '" /> <div class="caption"> Current  </div> </div> <div class="image"> <img class="passed" src="' + res.image + '" /> <div class="caption"> Baseline   </div> </div> </div> </div>'
```
Comment (sglanzer) [inline vendor/VisualAcceptance.js]: Changed

File: vendor/phantomjs-launcher.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
+'use strict'
+var system = require('system')
+var fs = require('fs')
+var page = require('webpage').create()
+var url = system.args[1]
+page.viewportSize = {
+  width: 1024,
+  height: 768
+}
```
Comment (sglanzer) [inline vendor/phantomjs-launcher.js]: Is there an issue raised to make these configurable?  I remember talking with Seena about that before

File: vendor/phantomjs-launcher.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      //Asynchronously
+      function(config, data, callback) {
+          callback(null)
+        
+        setTimeout( function (params) {
+          var status = window.callPhantom({
+            command: 'exit',
+            reason:  'transport close'
+          })
+        }, 2000)
```
Comment (sglanzer) [inline vendor/phantomjs-launcher.js]: Comment to explain this oddness to people not in the know

File: vendor/phantomjs-launcher.js (hunk 7 above)
Comment (EWhite613) [inline vendor/phantomjs-launcher.js]: @sglanzer Those are just the size of the `window`. The multiple size thing changes the width/height of the `.ember-testing-container`

File: tests/integration/simple-test.js (hunk 3 above)
Comment (EWhite613) [inline tests/integration/simple-test.js]: @sglanzer I'd like to keep that log. Good for a build to see what ResembleJS is returning.

File: vendor/VisualAcceptance.js (hunk 4 above)
Comment (EWhite613) [inline vendor/VisualAcceptance.js]: Will get fixed with this https://github.com/ciena-blueplanet/ember-cli-visual-acceptance/pull/44

File: vendor/VisualAcceptance.js (hunk 6 above)
Comment (EWhite613) [inline vendor/VisualAcceptance.js]: Will get fixed with https://github.com/ciena-blueplanet/ember-cli-visual-acceptance/pull/44

**Pull request merged**